decoder/src/lib.rs

```diff
@@ -1025,16 +1025,38 @@ fn format_args_real(
                     Arg::IStr(x) => format_str(x, hint, &mut buf)?,
                     Arg::Format { format, args } => buf.push_str(&format_args(format, args, hint)),
                     Arg::FormatSlice { elements } => {
-                        buf.write_str("[")?;
-                        let mut is_first = true;
-                        for element in elements {
-                            if !is_first {
-                                buf.write_str(", ")?;
+                        match hint {
+                            // Filter Ascii Hints, which contains u8 byte slices
+                            Some(DisplayHint::Ascii)
+                                if elements.iter().filter(|e| e.format == "{=u8}").count() != 0 =>
+                            {
+                                let vals = elements
+                                    .iter()
+                                    // Assumption: Arguments for u8 has always only 1 element
+                                    .filter_map(|e| match e.args[0] {
```

Comment:
I would suggest using `map` instead of `filter_map` and panicking if `arg` is not a single `Arg::Uxx`. Because: if for some reason (i.e. a bug) the program reaches that state then the current logic will silently ignore the errors and print the wrong thing. With a panic the bug will be immediately exposed.

Reply:
Converted both assumptions to panics, if they are not true.